framework/src/constraints/EqualValueEmbededConstraint.C

```diff
@@ -0,0 +1,368 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "EqualValueEmbededConstraint.h"
+#include "FEProblem.h"
+#include "DisplacedProblem.h"
+#include "AuxiliarySystem.h"
+#include "SystemBase.h"
+#include "Assembly.h"
+#include "MooseMesh.h"
+#include "Executioner.h"
+#include "AddVariableAction.h"
+
+#include "libmesh/string_to_enum.h"
+#include "libmesh/sparse_matrix.h"
+
+registerMooseObject("MooseApp", EqualValueEmbededConstraint);
+
+template <>
+InputParameters
+validParams<EqualValueEmbededConstraint>()
+{
+  MooseEnum orders(AddVariableAction::getNonlinearVariableOrders());
+  InputParameters params = validParams<NodeElemConstraint>();
+  params.addClassDescription("This is a constraint enforcing overlapping portions of two blocks to "
+                             "have the same variable value");
+  params.set<bool>("use_displaced_mesh") = false;
+  params.addParam<std::string>("formulation", "kinematic", "The formulation");
+  params.addParam<Real>(
+      "penalty",
+      1e8,
+      "The penalty to apply.  This can vary depending on the stiffness of your materials");
```

Comment:
I'd make this required because this can vary widely by problem. Use this for this description string:
"Penalty parameter used in constraint enforcement for kinematic and penalty formulations."